subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -224,56 +325,222 @@ static int validate_ak_order(void)
 	return 0;
 }
 
-static int fp_settings_validate(void)
+static int fp_settings_suffixed_name_gen(char *name, size_t name_max_size, const char *name_prefix,
+					 uint8_t index)
+{
+	int n;
+
+	n = snprintf(name, name_max_size, "%s%u", name_prefix, index);
+	__ASSERT_NO_MSG(n < name_max_size);
+	if (n < 0) {
+		return n;
+	}
+
+	return 0;
+}
+
+static int ak_name_gen(char *name, size_t name_len, uint8_t index)
+{
+	__ASSERT_NO_MSG(name_len == SETTINGS_AK_NAME_MAX_SIZE);
+	return fp_settings_suffixed_name_gen(name, name_len, SETTINGS_AK_FULL_PREFIX, index);
+}
+
+static int fp_bond_info_name_gen(char *name, size_t name_len, uint8_t index)
+{
+	__ASSERT_NO_MSG(name_len == SETTINGS_BOND_INFO_NAME_MAX_SIZE);
+	return fp_settings_suffixed_name_gen(name, name_len, SETTINGS_BOND_INFO_FULL_PREFIX, index);
+}
+
+static size_t fp_bond_info_idx_get(struct fp_bond_info *bond)
+{
+	return bond - fp_bonds;
+}
+
+static int fp_bond_info_settings_delete(struct fp_bond_info *bond)
 {
-	uint8_t id;
-	int first_zero_idx = -1;
 	int err;
+	char name[SETTINGS_BOND_INFO_NAME_MAX_SIZE];
 
-	if (settings_set_err) {
-		return settings_set_err;
+	err = fp_bond_info_name_gen(name, sizeof(name), fp_bond_info_idx_get(bond));
+	if (err) {
+		return err;
 	}
 
-	for (size_t i = 0; i < ACCOUNT_KEY_CNT; i++) {
-		id = ACCOUNT_KEY_METADATA_FIELD_GET(account_key_metadata[i], ID);
-		if (id == 0) {
-			first_zero_idx = i;
-			break;
+	err = settings_delete(name);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+static int bond_remove(struct fp_bond_info *bond, bool bt_stack_bond_remove)
+{
+	int err;
+
+	__ASSERT_NO_MSG(!bt_addr_le_eq(&bond->nv.addr, BT_ADDR_LE_ANY));
+
+	if (bt_stack_bond_remove) {
+		__ASSERT_NO_MSG(bt_request_cb);
+
+		err = bt_request_cb->bond_remove(&bond->nv.addr);
+		if (err) {
+			if (err == -ESRCH) {
+				LOG_DBG("Bond not found in the Bluetooth stack. "
+					"Continuing the removal operation");
```

Comment:
That's what the log is for. If I'd leave only the `"Bond not found in the Bluetooth stack. "` message it could give the false impression that the removal operation is aborted if someone calls `bond_remove` with `bt_stack_bond_remove` set to true if the bond is not present in the BT stack.

Reply:
Ok, we can keep it as is